src/card/collapsible/CollapsibleCard.tsx

```diff
@@ -0,0 +1,50 @@
+import "./_collapsible-card.scss";
+
+import React, {useState, useRef, useEffect} from "react";
+import classNames from "classnames";
+
+import Card from "../Card";
+
+export interface CollapsibleCardProps {
+  header: React.ReactNode;
+  initialIsOpen?: boolean;
+  children?: React.ReactNode;
+  onChange?: (state: "open" | "close") => void;
+}
+
+function CollapsibleCard({
+  header,
+  initialIsOpen = false,
+  children,
+  onChange
+}: CollapsibleCardProps) {
+  const [isOpen, setIsOpen] = useState(initialIsOpen);
```

Comment:
I think, we can keep this on the parent component. So that, it will be a controlled component. Let's think, we have global `appState` and some `CollapsibleCard` components. These `CollapsibleCard` components will affect other user behaviors. In this case, we would prefer to manage the states on the parent level.